openedx/core/djangoapps/catalog/docs/decisions/001-programs-cache.rst

```diff
@@ -0,0 +1,83 @@
+Programs Cache in LMS
+_____________________
+
+Status
+======
+Pending
+
+Context
+=======
+The LMS in edx-platform has a limited understanding of the structure and content of programs, as they are stored authoritatively in the Discovery service's SQL database. However, certain features of the LMS, such as displaying a learner’s programs on their dashboard, require access to the programs’ data. To fulfill this need, the LMS has the JSON responses from Discovery’s /api/v1/programs endpoint stored in memcached. The LMS uses the cache to understand what program or programs a course (run) is in. The contents of the cache are exposed through the ``openedx.core.djangoapps.catalog`` app. This system is what we are referring to when we say "the programs cache” or simply “the cache”.
+
+This cache is populated by calling LMS’s cache_programs management command.
+The cache is regularly refreshed via an external process.
+By design, the cache does not populate itself by calling out to the Discovery API upon a cache miss.
+
+Below is a (not-necessarily-exhaustive) list of features that depend on the programs cache:
+
+* The learner dashboard shows “Related Programs” under the cards of courses that are in programs.
+* The programs progress page uses the programs cache.
+* The creation of external IDs for users enrolled in courses within MicroBachelors programs requires knowledge of programs.
+* The program_enrollments system, which Masters uses for institution-managed enrollments in degree programs, uses the programs cache for validation of when creating associations between program enrollments and course enrollments.
+* The LMS management commands that populate program credentials need knowledge of programs.
+* Temporary Experiments often utilize the programs cache.
+* Courseware will reference programs information in order to show in-course recommendations.
+
+Issues
+======
+Correctness & Resiliency
+------------------------
+
+Because the cache is held in volatile memory, any crash or restart of the server holding the programs cache can surface these issues. While the cache_programs command can be run immediately afterwards, this is not a foolproof solution:
```

Comment:
`restart of the server` I think we should be explicit and say: `restart of the memcache server`